Fix typo in alerting_framework_health field name

TPXP opened the pull request:
## Summary

Currently, the `api/alerting/_health` endpoints returns data like this
```js
{ is_sufficiently_secure: true,
  has_permanent_encryption_key: true,
  alerting_framework_heath:
   { decryption_health: { status: 'ok', timestamp: '2021-12-06T14:23:35.900Z' },
     execution_health: { status: 'ok', timestamp: '2021-12-06T14:23:35.900Z' },
     read_health: { status: 'ok', timestamp: '2021-12-06T14:23:35.900Z' } } }
```

The `alerting_framework_heath` is misspelled, it should read `alerting_framework_health` (the L at the end is missing, while it's in the documentation https://www.elastic.co/guide/en/kibana/current/get-alerting-framework-health-api.html).

### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Previous code may rely on the misspelt `alerting_framework_heath` value | High | Low | Affected code would be monitoring code, thus production impact is negligible |

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

This can be considered as a breaking change and should be labeled accordingly I guess 😄

kibanamachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

YulNaumenko reviewed (approved):
LGTM

elasticmachine commented:
Pinging @elastic/kibana-alerting-services (Team:Alerting Services)

YulNaumenko commented:
jenkins retest this please

YulNaumenko commented:
jenkins retest this please

YulNaumenko reviewed (changes requested):
Need to change the the approval to reject, because according to our `Backward compatibility and breaking changes ` requirements we need to deprecate / add new corrected field.

YulNaumenko commented:
@TPXP would you mind to do the changes around he [comment](https://github.com/elastic/kibana/pull/120496#pullrequestreview-824751681) about backward compatibility? Please let me know if you need any help.

TPXP commented:
Hello, I'll give it a try this weekend

TPXP commented:
Rebased and updated to make sure the API routes still return the older field name, and updated tests. I think this won't break existing code now

YulNaumenko commented:
@elasticmachine merge upstream

YulNaumenko commented:
@TPXP thank you for doing the changes! One more thing I want to ask you: we need to let our users know that old version is deprecated, so just a tiny extension of the docs:
[/docs/api/alerting/health.asciidoc ](https://github.com/elastic/kibana/blob/main/docs/api/alerting/health.asciidoc) 

```
| deprecated::`alerting_framework_heath` 
| This state property has a typo. This is  three substates that identify the health of the alerting framework API: `decryption_health`, `execution_health`, and `read_health`.
```

YulNaumenko commented:
jenkins retest this please

YulNaumenko commented:
@elasticmachine merge upstream

TPXP commented:
:heavy_check_mark:  Updated the docs in the PR. I've also added a `_deprecated` field in the server response to make it easier to understand why two seemingly identical fields exist by simply looking at the response.

BTW, Thanks for your reactivity, I really appreciate it!

YulNaumenko reviewed (approved):
LGTM

YulNaumenko commented:
jenkins retest this please

tylersmalley commented:
Buildkite, test it

YulNaumenko commented:
Buildkite, test it

YulNaumenko commented:
Buildkite, test it

YulNaumenko commented:
Buildkite, test it

YulNaumenko commented:
Buildkite, test it

YulNaumenko commented:
Buildkite, test it

YulNaumenko commented:
jenkins retest this please

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.
